godot-kotlin/godot-library/src/nativeCore/kotlin/godot/core/type/CoreType.kt

```diff
@@ -8,6 +8,26 @@ internal interface CoreType {
     fun setRawMemory(mem: COpaquePointer)
 }
 
+/**
+ * WARNING: Do not inherit from this class, it is only there for the coretypes not entirely reimplemented in Kotlin.
+ */
+abstract class NativeCoreType<C : CStructVar> internal constructor() : CoreType {
```

Comment:
Just a suggestion: As this class is not intended to be inherited, i wonder if we should make that clear inside the className and not only in the kdoc. Something like `_NativeCoreType` or `InternalNativeCoreType`. Just a suggestion. I'm also fine with leaving it with the kdoc

Reply:
I'll probably wait for all the CoreTypes to be implemented and then we will refactor details like this.